projects/plugins/jetpack/extensions/blocks/subscriptions/controls.js

```diff
@@ -259,7 +259,10 @@ export default function SubscriptionControls( {
 
 				<WidthControl
 					width={ buttonWidth }
-					onChange={ newButtonWidth => setAttributes( { buttonWidth: newButtonWidth } ) }
+					onChange={ newButtonWidth => {
+						setAttributes( { buttonWidth: newButtonWidth } );
+						setAttributes( { buttonOnNewLine: newButtonWidth === '100%' } );
```

Comment:
Can these be combined into one call? Also, I think it could also be better if you use the original value or the new width check. Currently when I set the button to be on a newline then toggle the sizes below 100 it repeatedly turns off my previous setting.

```suggestion
						setAttributes( {
							buttonWidth: newButtonWidth,
							buttonOnNewLine: buttonOnNewLine || newButtonWidth === '100%',
						} )
```